cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/common/BatchFileFilter.java

```diff
@@ -40,12 +44,21 @@
   //length of 'YYYY-MM-dd-HH-mm"
   private static final int DATE_LENGTH = 16;
   private static final SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd-HH-mm");
+  private static final Gson GSON = new Gson();
+  private static final Type ARRAYLIST_DATE_TYPE  = new TypeToken<ArrayList<Date>>() { }.getType();
   private boolean useTimeFilter;
 
   private Pattern regex;
   private String pathName;
   private String lastRead;
   private Date prevMinute;
+  /*
+   * dateRangesToRead is an odd length ArrayList of Dates. The non terminal elements are tuples of Dates that
```

Comment:
Perhaps, also describe what the last element signifies? Or did I miss that somewhere?